pkgs/by-name/hy/hydrapaper/package.nix

```
@@ -0,0 +1,66 @@
+{ lib,
+  stdenv,
+  fetchFromGitLab,
+  meson,
+  ninja,
+  python3,
+  blueprint-compiler,
+  gobject-introspection,
+  pandoc,
+  libadwaita,
+  glib,
+  pkg-config,
+  dbus,
+  appstream,
+  wrapGAppsHook4
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hydrapaper";
+  version = "3.3.2";
+
+  src = fetchFromGitLab {
+    owner = "gabmus";
+    repo = "HydraPaper";
+    rev = finalAttrs.version;
+    hash = "sha256-IDaM8bM/0KH9h59523WqLKe400V5lLNyJ4faPf980Ro=";
+  };
+
+  # wrapGAppsHook4 propogates gtk4 -- which provides gtk4-update-icon-cache instead
+  preInstall = ''
+    substituteInPlace /build/source/meson_post_install.py \
+      --replace-fail gtk-update-icon-cache gtk4-update-icon-cache
+  '';
+
+  nativeBuildInputs = [
+    blueprint-compiler
+    gobject-introspection
+    meson
+    pandoc
+    ninja
+    glib
+    pkg-config
+    appstream
+    wrapGAppsHook4
+  ];
+
+  buildInputs = [
+    dbus
+    glib
+    libadwaita
+    (python3.withPackages (ps: with ps; [
+      dbus-python
+      pygobject3
+      pillow
+    ]))
+  ];
+
```

```suggestion
preFixup = ''
  gappsWrapperArgs+=(
    --prefix PATH : ${lib.makeBinPath [ glib xdg-user-dirs ]}
  )
'';
```

Didn't wrap `swaymsg` as it is part of `sway` package. We'd assume it is there.